### DISCUSSION TITLE
Understand if app is running in desktop mode.

### DISCUSSION THREAD

**Discussion by andreapesci** [tauri-apps/tauri#6941] [Q&A] [answered]
Hello, small background to better understand my question.

I have an Angular project which is published as a standard website.

I used Tauri to package it as a desktop app, and it worked perfectly fine and i'm very happy with the results. Now i would like to implement a new feature, available ONLY for the desktop app. Is it possible to be aware in which mode the app is running to enable/disable certain features?

**Comment by FabianLars** [comment, marked as answer]
If it's enough to check it at runtime you can check if `window,__TAURI__` exists. If your frontend supports a "conditional compilation"-like thing then you can check for any of the `TAURI_` env vars mentioned here https://tauri.app/v1/api/config#buildconfig.beforebuildcommand. For example in plain Vite i once used this plugin https://github.com/rollup/plugins/tree/master/packages/replace, which completely removes unused code from the dist files.

**Reply by andreapesci** [reply]
Thanks a lot, will see if this solves completely my problem (i think it does!).

Thanks again!
